executor/write_test.go

```diff
@@ -2458,6 +2458,25 @@ func (s *testSuite4) TestDeferConstraintCheckForInsert(c *C) {
 	tk.MustExec(`update t set i = 2 where i = 1;`)
 	tk.MustExec(`commit;`)
 	tk.MustQuery(`select * from t;`).Check(testkit.Rows("2"))
+
+	tk.MustExec(`drop table t; create table t (id int primary key, v int unique);`)
```

Comment:
```sql
create table t (i int key);
insert into t values (1), (2);
begin;
update t set i = 2 where i = 1; 
insert into t values (1) on duplicate key update i = i + 1;
commit/rollback;
```
Could you add a case like above?

Reply:
Done.
As we do not use the optimization, both `update` and `insert on duplicate` reports an error immediately.
https://github.com/pingcap/tidb/pull/11720/files#diff-84f3439ff795a2ffd72177b335c225d3R170